pbp/pbp.py

```diff
@@ -1,38 +1,17 @@
 #!/usr/bin/env python2
+import os
 import pysodium as nacl, scrypt # external dependencies
 import getpass, sys, struct
-from utils import b85encode, b85decode, lockmem, inputfd, outputfd, inc_nonce
 from SecureString import clearmem
-import chaining, publickey, ecdh
-import os
+from . import chaining, publickey, ecdh
+from .utils import b85encode, b85decode, lockmem, inputfd, outputfd, inc_nonce
+from .publickey import getkey
+from .config import scrypt_salt
 
 ASYM_CIPHER = 5
 BLOCK_CIPHER = 23
 BLOCK_SIZE = 32*1024
 
-defaultbase='~/.pbp'
-scrypt_salt = 'qa~t](84z<1t<1oz:ik.@IRNyhG=8q(on9}4#!/_h#a7wqK{Nt$T?W>,mt8NqYq&6U<GB1$,<$j>,rSYI2GRDd:Bcm'
-
-def getkey(size, pwd='', empty=False, text=''):
```

Comment:
why did you move this to publickey?

Reply:
To avoid circular imports, `getkey` is used by `publickey.py` and `pbp.py` imports `publickey.py`. This is not a problem in Python 2 because of relative imports. In Python 3.4, an error is raised for circular imports since it requires explicit relative imports but also does not support resolving circular dependencies (as Python 3.5+ does).

Reply:
Circular imports is also why `defaultbase` and `scrypt_salt` were moved to `config.py`, I tried to keep refactoring to a minimum, but was needed to resolve circular imports.